indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -546,6 +553,41 @@ public TaskStatus runTask(final TaskToolbox toolbox)
     }
   }
 
+  private void emitUnparseableEvents(TaskToolbox toolbox)
+  {
+    String taskId = getId();
+    for(Map.Entry<String, Object> unparseableEvents : getTaskCompletionUnparseableEvents().entrySet()) {
+      if (unparseableEvents.getValue() instanceof List) {
```

Comment:
Just curious, is this always of List type and never an object array ?

Reply:
I'm not sure need to dig deeper into the code to figure that out. 
Would be smarter to cast it to Iterable and only iterate over that in case it is a set

Reply:
It should always be a List